internal/kubernetes/client.go

```diff
@@ -0,0 +1,126 @@
+package kubernetes
+
+import (
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/cli-runtime/pkg/resource"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/restmapper"
+	"k8s.io/kubectl/pkg/util"
+)
+
+const (
+	// Default cache directory.
+	cacheDir       = "/var/kube/cache"
+	defaultTimeout = 180 * time.Second
+)
+
+var (
+	ttl = 10 * time.Minute
+)
+
+type client struct {
+	c      dynamic.Interface
+	config *rest.Config
+	mapper *restmapper.DeferredDiscoveryRESTMapper
+	log    *zap.SugaredLogger
+}
+
+// apply will do create/patch of manifest
+func (c *client) apply(u *unstructured.Unstructured) error {
+	gvk := u.GroupVersionKind()
+	restMapping, err := c.mapper.RESTMapping(gvk.GroupKind(), gvk.Version)
+	if err != nil {
+		return err
+	}
+
+	gv := gvk.GroupVersion()
+	c.config.GroupVersion = &gv
+
+	restClient, err := newRestClient(*c.config, gv)
+	if err != nil {
+		return err
+	}
```

Comment:
In the future, I'm thinking we should explore maybe having a reusable TCP client rather than needing to start up a new connection with each apply.

Reply:
sure, I'll add TODO there, need to find some approach as it's generating restClient based on groupVersion of manifest.

Reply:
thanks, I just wrote this issue: https://github.com/numaproj-labs/numaplane/issues/52